include/swift/SIL/SILModule.h

```diff
@@ -88,6 +88,37 @@ enum class SILStage {
   Lowered,
 };
 
+typedef llvm::function_ref<void()> ActionCallback;
```

Comment:
Does this really belong inside the SILModule? Seems like a lot of boilerplate for just one callback. Why not split up the optimizer entry point into two entry points instead, and have the frontend serialize the module in between calling the two entry points?